### PR TITLE
Add Footer component

### PR DESCRIPTION
**Pull request by joeytepp**
## Proposed Change

Fixes: #114

Add a `Footer` component to the QHacks 2019 Website 🎉 🎉 🎉 🎉 

### How did you do this?
Made a `Footer` component, added it to the index page updated tests

### Why are you choosing this approach?
To get this website to completion 🔥 🔥 🔥  (only sponsors left to go!!)

### Any open questions you want to ask code reviewers?
No questions but apologies for the huge diff (a lot of new assets in this PR)

### List of changes:

  - Add `Footer` component
  - Add component to index page
  - Update and add new tests

## Pull Request Checklist:

  - [x] My submission passes all tests.
  - [x] I have lint my code locally prior to submission.
  - [x] I have written new tests for my core changes (where applicable).
  - [x] I have referenced all useful information (issues, tasks, etc) in the pull request.

### CODE REVIEW

**Comment by RobertWSaunders** [comment]
Deploy preview for *qhacks-website* ready!

Built with commit 15dca2c5027fd8025b415eb9aed77a561019e73d

https://deploy-preview-124--qhacks-website.netlify.com

File: src/components/Footer.js [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+            Copyright © 2018 QHacks
+          </p>
+        </div>
+      </div>
+      <div css={{ [largerScreens]: { display: "none" } }}>
+        <p css={{ ...finePrint, paddingBottom: "6px" }}>
+          Queen&#39;s University | 99 University Ave, Kingston, ON
+        </p>
+        <p css={{ ...finePrint, paddingBottom: "6px" }}>
+          {"< />"} with ♡ by Queen&#39;s Students
```
Comment (joeytepp) [inline src/components/Footer.js]: Brings a tear to my eye 😢 😂 😢 😂

File: src/__tests__/pages/Index/__snapshots__/index.test.js.snap [hunk 2]
```diff
@@ -50,5 +50,6 @@ exports[`IndexPage renders correctly 1`] = `
   <HackerTestimonials />
   <FAQs />
   <Podcast />
+  <Footer />
```
Comment (rosslh) [inline src/__tests__/pages/Index/__snapshots__/index.test.js.snap]: :heart_eyes:

File: src/components/Footer.js (hunk 1 above)
Comment (rosslh) [inline src/components/Footer.js]: You matched the mocks, but i feel like a solid heart would look better (I know, totally minor :slightly_smiling_face:  )

```suggestion
          {"< / >"} with 🖤 by Queen&#39;s Students
```

Comment (joeytepp) [reply src/components/Footer.js]: ^ nice use of the new GH feature 🌟

Comment (joeytepp) [reply src/components/Footer.js]: I'm impartial @RobertWSaunders what are your thoughts?

File: src/components/Footer.js [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+          <p
+            css={{
+              ...finePrint,
+              paddingBottom: "32px",
+              paddingTop: "132px",
+              textAlign: "right",
+              [smallerScreens]: { display: "none" }
+            }}
+          >
+            Copyright © 2018 QHacks
```
[the remark this answers is not in the document]
Comment (joeytepp) [reply src/components/Footer.js]: That's where I had it originally, but moved it here to match the mocks. Will move it back

File: src/components/Footer.js [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+import facebook from "../assets/img/facebook.svg";
+import instagram from "../assets/img/instagram.svg";
+import mail from "../assets/img/mail.svg";
+import twitter from "../assets/img/twitter.svg";
+
+const navigateLinks = [
+  { label: "QHacks 2018", url: "https://2018.qhacks.io" },
+  { label: "Local Hack Day", url: "https://localhackday.mlh.io/" },
+  { label: "QHacks Blog", url: "https://medium.com/qhacks" },
+  { label: "QHacks Dashboard", url: "https://aqq.qhacks.io" },
```
[the remark this answers is not in the document]
Comment (joeytepp) [reply src/components/Footer.js]: Took me a while to see what the typo was, I'll fix it anyway ahahaha

File: src/assets/img/mentors.svg [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
... 0-.782 5.226 18.57 18.57 0 0 0 .67 3.334c.178.573.4 1.133.665 1.672a4.98 4.98 0 0 0 .507.78H20.146a8.481 8.481 0 0 0-1.888-3.401l.397-.138c.4-.14.703-.246.855-.293 2.297-.716 3.445-1.753 3.414-3.083a.534.534 0 0 0-.548-.52.534.534 0 0 0-.523.545c.013.549-.439 1.348-2.664 2.042-.158.05-.473.159-.889.304-.384.134-.856.298-1.374.474h-.003a44.739 44.739 0 0 1-4.601 1.382.948.948 0 0 1-1.105-.714 2.05 2.05 0 0 1-.13-.94c.015-.11.04-.22.065-.33.11-.37.127-.76.05-1.138a1.761 1.761 0 0 0-.987-1.09l-.115-.065c.057-.116.126-.227.205-.33a.53.53 0 0 0-.103-.756 25.794 25.794 0 0 1-.655-.497c.027-.05.058-.104.087-.153.082-.145.176-.311.267-.502.172-.373.222-.789.144-1.19-.125-.653-.752-.971-1.21-1.204a3.11 3.11 0 0 1-.446-.256 2.742 2.742 0 0 1-.315-.254c.001-.08.085-.275.497-.774a21.445 21.445 0 0 0 2.448-3.955 1.53 1.53 0 0 0 .21-.722 1.854 1.854 0 0 0-.785-1.332c-.217-.197-.421-.383-.444-.546a.923.923 0 0 1 .086-.374c.087-.232.19-.46.292-.687.219-.447.393-.914.52-1.395l.045-.194c.114-.543.263-1.08.446-1.604a9.035 9.035 0 0 1 1.921-3.117c1.038-1.089 2.609-1.83 4.8-2.27a35.172 35.172 0 0 1 5.026-.716c3.031-.151 5.632.753 8.436 2.94a7.683 7.683 0 0 1 2.832 4.03c.375 1.372.55 2.791.52 4.213z"/>
```
[the remark this answers is not in the document]
Comment (joeytepp) [reply src/assets/img/mentors.svg]: I'll see if i can fix it

File: src/components/Footer.js [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+];
+
+const Footer = () => (
+  <div css={{ backgroundColor: "#11213f", color: "#f1f1f1" }}>
+    <div
+      css={{
+        display: "grid",
+        gridTemplateColumns: "auto auto auto",
+        paddingTop: "80px",
+        " h2": {
```
[the remark this answers is not in the document]
Comment (joeytepp) [reply src/components/Footer.js]: It's necessary for `glamor` yea

File: src/components/Footer.js [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+        </div>
+      </div>
+      <div css={{ [largerScreens]: { display: "none" } }}>
+        <p css={{ ...finePrint, paddingBottom: "6px" }}>
+          Queen&#39;s University | 99 University Ave, Kingston, ON
+        </p>
+        <p css={{ ...finePrint, paddingBottom: "6px" }}>
+          {"< />"} with ♡ by Queen&#39;s Students
+        </p>
+        <p css={{ ...finePrint, paddingBottom: "32px" }}>
```
[the remark this answers is not in the document]
Comment (joeytepp) [reply src/components/Footer.js]: Yup was the only way to get it to align properly. Only one will be visible at a time though

File: src/components/Footer.js [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+              marginLeft: "auto",
+              marginRight: "auto",
+              gridTemplateColumns: "1fr 1fr 1fr 1fr",
+              maxWidth: "196px",
+              paddingTop: "0px",
+              paddingBottom: "60px"
+            }
+          }}
+        >
+          {socialLinks.map(({ icon, label, url }) => (
```
[the remark this answers is not in the document]
Comment (joeytepp) [reply src/components/Footer.js]: Sure thing 👌

File: src/components/SignUpForm.js [hunk 9]
```diff
@@ -1,46 +1,56 @@
...
+          width: "300px",
+          minWidth: "50%",
+          paddingLeft: "16px",
+          paddingRight: "16px",
+          height: "48px",
+          ...fixInputAlignment
+        }}
+      />
+      <ActionButton
+        backgroundColor={props.backgroundColor || "white"}
```
[the remark this answers is not in the document]
Comment (joeytepp) [reply src/components/SignUpForm.js]: There weren't any props for this component to begin with, so thought it made sense to do this to default to white

File: src/components/SignUpForm.js [hunk 10]
```diff
@@ -1,46 +1,56 @@
...
-      foregroundColor="#00205b"
-      type="rect"
-      style={buttonStyles}
     >
-      Sign Up
-    </ActionButton>
-  </div>
-);
+      <input
+        placeholder="Enter your email address"
```
[the remark this answers is not in the document]
Comment (joeytepp) [reply src/components/SignUpForm.js]: Yea this was actually super annoying but when I changed the font size from `18px` to `14px` the input and button became misaligned. Had to play around with the padding to get them back in line. I'll give it another look and see if I can align them and keep the vertical text alignment

File: src/components/Footer.js (hunk 1 above)
Comment (RobertWSaunders) [reply src/components/Footer.js]: Nice, I wonder if you did it for like 20 things, would it be 20 separate commits, hmmm.

File: src/assets/img/mentors.svg (hunk 5 above)
[the remark this answers is not in the document]
Comment (RobertWSaunders) [reply src/assets/img/mentors.svg]: Great, should just be spacing!

File: src/components/Footer.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (RobertWSaunders) [reply src/components/Footer.js]: Lol, yeah it looks very similar but is very different.

File: src/components/Footer.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (RobertWSaunders) [reply src/components/Footer.js]: Nice, will resolve this!

File: src/components/Footer.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (RobertWSaunders) [reply src/components/Footer.js]: Yeah, I don't agree with the mocks, but good job to copy them.

File: src/components/Footer.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (RobertWSaunders) [reply src/components/Footer.js]: Sounds good! Will resolve this!

File: src/components/Footer.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (RobertWSaunders) [reply src/components/Footer.js]: Noice!

File: src/components/SignUpForm.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (RobertWSaunders) [reply src/components/SignUpForm.js]: Fair enough! Will mark this as resolved!

File: src/components/Footer.js (hunk 1 above)
Comment (joeytepp) [reply src/components/Footer.js]: Seems like it is a separate commit

Comment (joeytepp) [reply src/components/Footer.js]: Will make this change tho!

Comment (joeytepp) [reply src/components/Footer.js]: UPDATE: the heart shows up as black, so back to the original it is

**Comment by joeytepp** [comment]
@RobertWSaunders updated!

File: src/components/Footer.js (hunk 1 above)
Comment (RobertWSaunders) [reply src/components/Footer.js]: Sounds good, marking this as resolved!

**Comment by RobertWSaunders** [comment]
Ah, it appears that Coveralls is back! Did you change anything @joeytepp?

File: src/assets/img/mentors.svg [hunk 11]
```diff
@@ -1,7 +1,7 @@
...
```
Comment (RobertWSaunders) [inline src/assets/img/mentors.svg]: Can we fix the spacing here?

File: src/components/SignUpForm.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (RobertWSaunders) [reply src/components/SignUpForm.js]: This looks better!

**Comment by joeytepp** [comment]
Seems like coveralls just went away for a while, didn't change anything maybe outages lol

**Comment by RobertWSaunders** [comment]
This is good to go! It appears statuses aren't being reported so I've disabled it for you.

**Pull request merged**